models/DataObject/Data/ElementMetadata/Dao.php

```diff
@@ -43,7 +43,7 @@ public function save(DataObject\Concrete $object, string $ownertype, string $own
             $data = $dataTemplate;
             $data['column'] = $column;
             $data['data'] = $this->model->$getter();
-            Helper::upsert($this->db, $table, $data, $this->getPrimaryKey($table));
+            Helper::upsert($this->db, $table, $data, [parent::UNIQUE_KEY_NAME]);
```

Comment:
to avoid bc breaks and if we cannot re-run the migration, it should be something like
```suggestion
            $key = check if parent::UNIQUE_KEY_NAME exists ?? $this->getPrimaryKey($table)
            Helper::upsert($this->db, $table, $data, $key);
```

Reply:
It seems it would be better to just merge in next minor and have to rerun the migration